plugin/pkg/auth/authenticator/token/oidc/oidc.go

```diff
@@ -95,69 +126,85 @@ func New(opts OIDCOptions) (*OIDCAuthenticator, error) {
 		TLSClientConfig: &tls.Config{RootCAs: roots},
 	})
 
-	hc := &http.Client{}
-	hc.Transport = tr
+	return &OIDCAuthenticator{
```

Comment:
it seems ok to let the client init async, but could we kick off a call to client() from here so in a normal case, the client setup delay doesn't affect the first user to log in? (also, an error might show up in the server log while an admin is still watching the log)

```
authenticator := &OIDCAuthenticator{ ... }

// initialize the client asynchronously
go func(){
  defer utilruntime.HandleCrash()
  authenticator.client()
}()

return authenticator, nil
```